api_app/analyzers_manager/file_analyzers/signature_info.py

```diff
@@ -22,41 +20,31 @@ def run(self):
             "verified": False,
             "corrupted": False,
         }
-        try:
-            command = ["osslsigncode", "verify", self.filepath]
-            p = Popen(command, stdin=DEVNULL, stdout=PIPE, stderr=PIPE)
-            (out, err) = p.communicate()
-            output = out.decode()
-
-            if p.returncode == 1 and "MISMATCH" in output:
-                results["checksum_mismatch"] = True
-            elif p.returncode != 0:
-                raise AnalyzerRunException(
-                    f"osslsigncode return code is {p.returncode}. Error: {err}"
-                )
-
-            if output:
-                if "No signature found" in output:
-                    results["no_signature"] = True
-                if "Signature verification: ok" in output:
-                    results["verified"] = True
-                if "Corrupt PE file" in output:
-                    results["corrupted"] = True
-            else:
-                raise AnalyzerRunException("osslsigncode gave no output?")
-
-        except SoftTimeLimitExceeded as e:
-            error_message = (
-                f"job_id:{self.job_id} analyzer:{self.analyzer_name} md5:{self.md5}"
-                f"filename: {self.filename}. Soft Time Limit Exceeded Error {e}"
+        command = ["osslsigncode", "verify", self.filepath]
+        p = Popen(command, stdin=DEVNULL, stdout=PIPE, stderr=PIPE)
+        (out, err) = p.communicate()
+        output = out.decode()
+
+        if p.returncode == 1 and "MISMATCH" in output:
+            results["checksum_mismatch"] = True
+        elif p.returncode != 0:
+            raise AnalyzerRunException(
+                f"osslsigncode return code is {p.returncode}. Error: {err}"
             )
-            logger.error(error_message)
-            self.report.errors.append(str(e))
-            self.report.status = self.report.Status.FAILED
-            self.report.save()
-            # we should stop the subprocesses...
-            # .. in case we reach the time limit for the celery task
-            if p:
-                p.kill()
+
+        if output:
+            if "No signature found" in output:
+                results["no_signature"] = True
+            if "Signature verification: ok" in output:
+                results["verified"] = True
+            if "Corrupt PE file" in output:
+                results["corrupted"] = True
+        else:
+            raise AnalyzerRunException("osslsigncode gave no output?")
+
+        # we should stop the subprocesses...
+        # .. in case we reach the time limit for the celery task
+        if p:
+            p.kill()
```

Comment:
Ah fuck, my bad I did not notice this.

Reply:
Sorry but we do need the try/catch block here.

Reply:
![image](https://user-images.githubusercontent.com/16389167/127625406-9c3e5d7e-ac96-4094-9518-ae730f55c7e0.png)

It's just that instead of all these lines, we can simply call `self._handle_exception(e)`.

Reply:
```
...
except SoftTimeLimitException as exc:
self._handle_exception(exc)
if p:
    p.kill()
```